category/traversable/instances.lean

```diff
@@ -0,0 +1,249 @@
+/-
+Copyright (c) 2018 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Simon Hudon
+
+Instances of `traversable` for types from the core library
+-/
+
+import category.traversable.basic
+import category.basic
+import category.functor
+import category.applicative
+
+universe variables u v
+
+open function
+
+section identity
+
+open function functor
+
+variables {f f' : Type u → Type u}
+variables [applicative f] [applicative f']
+
+def id.traverse {α β : Type u} (g : α → f β) : id α → f (id β) := g
+
+instance : traversable id :=
+⟨ @id.traverse ⟩
+
+variables [is_lawful_applicative f] [is_lawful_applicative f']
+variables {α β γ : Type u}
+
+lemma id.id_traverse (x : id α) :
+  id.traverse id.mk x = x :=
+by refl
```

Comment:
You should always use `rfl` to prove reflexivity theorems, because of the special support they get in `dsimp` and such. But I don't think we need any of these next few theorems, they should all be inlined since `id.traverse` will never be used directly like this.